bundle/jsky.app.ot/src/main/java/jsky/app/ot/viewer/SPElevationPlotPlugin.java

```diff
@@ -232,106 +212,122 @@ private void _setColorCode(String label) {
     }
 
 
-    /**  Return a custom LegendItemCollection based on the current settings  and set the item colors to match */
-    public LegendItemCollection getLegendItems() {
-        return _getLegendItems();
-    }
+    // Create a LegendItemCollection by:
+    // 1. Executing an optional preprocessing function (to precalculate some colors and create some default legend items); and
+    // 2. Executing an extractor for each selected observation, which produces an optional ID. If the ID is defined, create
+    //    a new legend item for it.
+    // We also ensure that IDs for LegendItems are unique so that we don't, for example, end up with multiple entries for
+    // "Band 4" if observations are labeled by band, or repeated priorities if observations are labeled by priority.
+    private LegendItemCollection createLegendItemsPreOpt(final Option<Function<TreeMap<String, Paint>, LegendItemCollection>> pref,
+                                                         final Function<ISPObservation, Option<String>> extractor) {
+        final Paint[] colors = new Paint[_selectedObservations.length];
```

Comment:
The ultimate goal is to get a color for each selected observation, and a collection of legend items describing those colors.